reporters/cloud-run-scripts/CloudRunnerHub.js

```diff
@@ -56,19 +55,30 @@ app.post("/", async (req, res) => {
 
     // Get the config file from the GCP bucket
     const configObject = await _fetchConfigObject(req.body.bucket, req.body.configFile);
-    console.log("configObject", configObject);
+
+    // Fetch the last block number this given config file queried the blockchain at.
+    const lastQueriedBlockNumber = await _getLastQueriedBlockNumber(req.body.configFile);
+
+    // Get the latest block number. The query will run from the last queried block number to the latest block number.
+    const latestBlockNumber = await _getLatestBlockNumber();
+
+    // Save the current latest block number to the remote config. This will be the used as the "lastQueriedBlockNumber"
+    // in the next iteration when the hub is called again.
+    await _saveQueriedBlockNumber(req.body.configFile, latestBlockNumber);
```

Comment:
this is a good point and something I thought about; what is the desired behavior in a failed case? do you want _all_ the scripts downstream that worked correctly to re-report on the blocks that because one of the internal scripts failed or would you like it to continue from where it left off? my preference was with the latter case but I'm happy to move it to afterward. Importantly the value in `latestBlockNumber` is pulled before the async calls so it doesn't really matter where it sits from a timing perspective.

Reply:
SG, I think that choice makes sense. Just wanted to bring it up in case you hadn't considered that option.